modules/external-auth-fields.adoc

```diff
@@ -0,0 +1,144 @@
+
+|`claimMappings.uid.claim`
+|An optional field for specifying the JWT token claim that is used in the mapping. The value of this claim will be assigned to the field in which this mapping is associated. Precisely one of claim or expression must be set. claim must not be specified when expression is set. When specified, claim must be at least 1 character in length and must not exceed 256 characters in length.
+
+|`claimMappings.uid.expression`
+|An optional field for specifying a CEL expression that produces a string value from JWT token claims.
+
+CEL expressions have access to the token claims through a CEL variable, `claims`, which is. `claims` is a map of claim names to claim values. For example, the `sub` claim value can be accessed as `claims.sub`. Nested claims can be accessed using dot notation (`claims.foo.bar`).
+
+Precisely one of `claim` or `expression` must be set. When specified, `expression` must be at least 1 character in length and must not exceed 4096 characters in length.
+
+|`claimMappings.username`
+|A name of the claim that should be used to construct usernames for the cluster identity. The default is `sub`.
+
+|`claimMappings.username.claim`
+|A JWT token claim to be used in the mapping.
+
+|`claimMappings.username.prefix`
+|How to prefix the value from the token in the result of the claim mapping.
+// TODO: This had no description, so I made one based on group
+
+|`claimMappings.username.prefix.prefixString`
+|A string to prefix the value from the token in the result of the claim mapping.
+// TODO: This had no description, so I made one based on group
+
+|`claimMappings.username.prefixPolicy`
+|How a prefix should apply. Allowed values are `""`, `NoPrefix`, or `Prefix`. By default, claims other than `email` will be prefixed with the issuer URL to prevent naming clashes with other plugins. Set to `NoPrefix` to disable prefixing.
+// TODO: Does this mean that if you set it to "" that it prefixes with the issuer URL? and if you want to set your own, you set it to Prefix? And set to NoPrefix to do no prefix at all?
+
+|`claimValidationRules`
+|Rules that are applied to validate token claims to authenticate users.
+
+|`claimValidationRules.requiredClaim`
+|Allows configuring a required claim name and its expected value.
+
+|`claimValidationRules.requiredClaim.claim`
+|A name of a required claim. Only claims with string values are supported.
+
+|`claimValidationRules.requiredClaim.requiredValue`
+|The required value for the claim.
+
+|`claimValidationRules.type`
+|The type of validation rule.
+// TODO not sure what to do with: enum: RequiredClaim
```

Comment:
@liouk @xingxingxia I see that it lists an enum of RequiredClaim, but I'm unsure what that means, if there's only one option listed.

Reply:
@bergerhoffer , yes only one option. I had a case (OCP-79806 Updating oidcProviders should work well in OCP) in our test case system where it covered `Test "claimValidationRules"` and `Update "requiredValue"`, which should clearly demo what it means.

Reply:
https://github.com/openshift/api/blob/a463917701ab4506585893739e576ffd1e6b924e/config/v1/types_authentication.go#L711-L720 might also help